framework/src/main/java/org/checkerframework/common/basetype/BaseTypeValidator.java

```diff
@@ -95,7 +95,7 @@ public boolean isValid(AnnotatedTypeMirror type, Tree tree) {
      * @return whether or not the top-level type should be checked
      */
     protected boolean shouldCheckTopLevelDeclaredType(AnnotatedTypeMirror type, Tree tree) {
-        if (type.getKind() != TypeKind.DECLARED) {
+        if (type.getKind() != TypeKind.DECLARED && !type.getKind().isPrimitive()) {
```

Comment:
Could you clarify the javadoc? It's confusing given the method name that this applies to anything other than declared types.
Can you call out explicitly what happens with ARRAY (and what else? intersection?)